packages/Chem/src/widgets/chem-substructure-filter.ts

```diff
@@ -149,28 +161,39 @@ export class SubstructureFilter extends DG.Filter {
     super.detach();
     if (this.column?.temp['chem-scaffold-filter'])
       this.column.temp['chem-scaffold-filter'] = null;
+    this.batchResultObservable?.unsubscribe();
   }
 
   applyFilter(): void {
-    if (this.bitset && !this.isDetached) {
-      this.dataFrame?.filter.and(this.bitset);
-      this.dataFrame?.rows.addFilterState(this.saveState());
-      this.column!.temp['chem-scaffold-filter'] = this.sketcher.getMolFile();
-      this.active = true;
+   // console.log(`in apply filter ${this.sketcher.getMolFile()}`)
+      if (this.bitset && !this.isDetached) {
+        this.dataFrame?.filter.and(this.bitset);
```

Comment:
The null-aware "?" operator makes no sense here, if dataFrame is null then it's still going to throw an exception when you apply .and to it. Same error in the following line as well.

this.dataFrame?.filter.and(this.bitset);

Reply:
Fixed